bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MediaRendererService.java

```diff
@@ -181,167 +219,153 @@ public String getUDN() {
-                "Master", "DesiredMute", Boolean.toString(newValue)));
-
-        updateResourceState("RenderingControl", "GetMute",
-                SamsungTvUtils.buildHashMap("InstanceID", "0", "Channel", "Master"));
-    }
-
-    private void setBrightness(Command command) {
-        int newValue;
-
-        try {
-            newValue = DataConverters.convertCommandToIntValue(command, 0, 100,
-                    Integer.valueOf(stateMap.getOrDefault("CurrentBrightness", "")));
-        } catch (NumberFormatException e) {
-            throw new NumberFormatException("Command '" + command + "' not supported");
+        if (command instanceof OnOffType) {
+            return Boolean.toString(command.equals(OnOffType.ON));
         }
-
-        updateResourceState("RenderingControl", "SetBrightness",
-                SamsungTvUtils.buildHashMap("InstanceID", "0", "DesiredBrightness", Integer.toString(newValue)));
-
-        updateResourceState("RenderingControl", "GetBrightness", SamsungTvUtils.buildHashMap("InstanceID", "0"));
+        return command.toString();
     }
 
-    private void setContrast(Command command) {
-        int newValue;
-
-        try {
-            newValue = DataConverters.convertCommandToIntValue(command, 0, 100,
-                    Integer.valueOf(stateMap.getOrDefault("CurrentContrast", "")));
-        } catch (NumberFormatException e) {
-            throw new NumberFormatException("Command '" + command + "' not supported");
-        }
-
-        updateResourceState("RenderingControl", "SetContrast",
-                SamsungTvUtils.buildHashMap("InstanceID", "0", "DesiredContrast", Integer.toString(newValue)));
-
-        updateResourceState("RenderingControl", "GetContrast", SamsungTvUtils.buildHashMap("InstanceID", "0"));
+    /**
+     * Parse Subscription Event from {@link String} which contains XML content.
+     * Parses all child Nodes recursively.
+     * If valid channel update is found, call onValueReceived()
+     *
+     * @param xml{@link String} which contains XML content.
+     */
+    public void parseEventValues(String xml) {
+        Utils.loadXMLFromString(xml, host).ifPresent(a -> visitRecursively(a));
     }
 
-    private void setSharpness(Command command) {
-        int newValue;
-
-        try {
-            newValue = DataConverters.convertCommandToIntValue(command, 0, 100,
-                    Integer.valueOf(stateMap.getOrDefault("CurrentSharpness", "")));
-        } catch (NumberFormatException e) {
-            throw new NumberFormatException("Command '" + command + "' not supported");
-        }
-
-        updateResourceState("RenderingControl", "SetSharpness",
-                SamsungTvUtils.buildHashMap("InstanceID", "0", "DesiredSharpness", Integer.toString(newValue)));
-
-        updateResourceState("RenderingControl", "GetSharpness", SamsungTvUtils.buildHashMap("InstanceID", "0"));
+    public void visitRecursively(Node node) {
```

Comment:
Method duplicated in `MainTVServerService`. Can logic be shared?

Reply:
I wanted to keep the Services like `MediaRenderer` and `MainTVServer` independent of each other, in case the logic had to be adjusted for each. I could move the method to the `Utils` class, but it didn't seem worth it for one method.

Reply:
OK.